File: lib/src/bin/tx_gen.py
import sys
import uuid
from timeit_decorator import timeit_sync
from crypto import PrivateKey
from lib.src.types.transaction import Transaction, TransactionOutput
from lib.src.types.block import INITIAL_REWARD


@timeit_sync(runs=5, workers=2)
def main():
    if len(sys.argv) != 2:
        print("Usage: tx_gen <tx_file>")
        sys.exit(1)
    path = sys.argv[1]
    private_key = PrivateKey.new_key()
    transaction = Transaction(
        inputs=[],
        outputs=[TransactionOutput(
            unique_id=uuid.uuid4(),
            value=INITIAL_REWARD * 10**8,
            public_key=private_key.public_key()
        )]
    )
    transaction.save(path)
    print("Done")


if __name__ == "__main__":
    main()

File: lib/src/bin/tx_gen.rs
use btclib::crypto::PrivateKey;
use btclib::types::{Transaction, TransactionOutput};
use btclib::util::Saveable;
use clap::Parser;
use uuid::Uuid;

#[derive(Parser)]
#[command(author, version, about, long_about = None)]
struct Cli {
    #[arg(short, long)]
    tx_file: String
}

fn main() {
    let cli = Cli::parse();

    let private_key = PrivateKey::new_key();
    let transaction = Transaction::new(
        vec![],
        vec![TransactionOutput {
            unique_id: Uuid::new_v4(),
            value: btclib::INITIAL_REWARD * 10u64.pow(8),
            public_key: private_key.public_key(),
        }],
    );

    transaction.save_to_file(cli.tx_file).expect("Failed to save transaction");
    println!("Saved");
}
